### DISCUSSION TITLE
Smart contract verification without the microservice

### DISCUSSION THREAD

**Discussion by human-77** [blockscout/blockscout#10571] [General]
Is there a way to verify a smart contract without the [smart-contract-verifier](https://github.com/blockscout/blockscout-rs/tree/main/smart-contract-verifier) microservice? If yes, how?

I have tried to run it without the microservice in my machine and the blockscout-frontend does enables the verification page, but, it always give a "verification failed" error when I try to actually verify it.

**Comment by ekawibisono** [comment]
hello @human-77 , What do you mean by contract verification without microservice ? Or do you have other ideas besides using microservices ?
